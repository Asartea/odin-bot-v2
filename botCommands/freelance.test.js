const command = require('./freelance');

describe('!freelance', () => {
  describe('regex', () => {
    it.each([
      ['!freelance'],
      [' !freelance'],
      ['!freelance @odin-bot'],
      ['@odin-bot !freelance'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['frlnce'],
      ['!frlence'],
      ['!ferlence'],
      ['! freelance'],
      ['!afreelance'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! !freelance'],
      ['Don\'t worry about !freelance'],
      ['Hey @odin-bot, !freelance'],
      ['!@odin-bot ^ !me !freelance !tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user!freelance'],
      ['it\'s about!freelance'],
      ['!freelanceanillusion'],
      ['!freelance!'],
      ['!freelance*'],
      ['!freelance...'],
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot();
    });
  });
});
