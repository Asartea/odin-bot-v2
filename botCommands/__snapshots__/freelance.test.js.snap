// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!freelance callback returns correct output 1`] = `"Here are some things that should be considered before getting into freelancing: https://discord.com/channels/505093832157691914/505093832157691916/928760451213443193 "`;
